### PR TITLE
Add cgroupv2 container id when possible

### PR DESCRIPTION
**Pull request by pawelchcki**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/telemetry.go [hunk 1]
```diff
@@ -136,3 +140,14 @@ func (m *telemetryMultiTransport) roundTrip(req *http.Request, endpoint *config.
...
+func ensureContainerID(ctx context.Context, h http.Header) {
+	if id := h.Get(headerContainerID); id != "" {
+		return
+	}
+	if id := GetContainerID(ctx, h); id != "" {
+		h.Set(headerContainerID, id)
+	} else {
+		metrics.Count("datadog.trace_agent.telemetry_proxy.no_container_id_found", 1, []string{}, 1)
+	}
+}
```
Comment (knusbaum) [inline pkg/trace/api/telemetry.go:153]: Unless we're going to use this other places, it may be clearer to just inline it.

**Review by knusbaum — CHANGES_REQUESTED** [review]
LGTM, just a test style comment.

File: pkg/trace/api/telemetry.go (hunk 1 above)
Comment (pawelchcki) [reply pkg/trace/api/telemetry.go]: Sounds good! inlined

**Comment by pawelchcki** [comment]
@knusbaum If you think its good, feel free to merge into your branch :bow:

File: releasenotes/notes/cgroup2-a3868da0e9ae5450.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    APM: Add support for cgroup2 via UDS.
```
Comment (maycmlee) [inline releasenotes/notes/cgroup2-a3868da0e9ae5450.yaml:11]:
```suggestion
    APM: Add support for cgroup2 using UDS.
```

**Review by maycmlee — APPROVED** [review]
small edit

**Comment by knusbaum** [comment]
Closed in favor of #13586

**Pull request closed without merging**